rabbitmq/producer.go

```diff
@@ -61,13 +61,19 @@ func (p *Producer) Publish(
 ) error {
 	select {
 	case rmqErr := <-p.closeCh:
-		p.logger.Warn(
-			"RMQ closed the connection",
-			zap.String("reason", rmqErr.Reason),
-			zap.Int("code", rmqErr.Code),
-			zap.Bool("recover", rmqErr.Recover),
-			zap.Bool("server", rmqErr.Server),
-		)
+		if rmqErr != nil {
+			p.logger.Warn(
+				"RMQ closed the connection",
+				zap.String("reason", rmqErr.Reason),
+				zap.Int("code", rmqErr.Code),
+				zap.Bool("recover", rmqErr.Recover),
+				zap.Bool("server", rmqErr.Server),
+			)
+		} else {
+			p.logger.Warn(
+				"RMQ closed the connection with nil rmqErr",
```

Comment:
while this seems intuitive, the log message doesn't say much why it happened, tho and whether this is good or bad behavior.

what's the "issue" you have atm?

Reply:
At the moment we have a case where we are trying to reconnect the client for rabbitMQ when connection fails, and while doing this we receive a nil rmqErr once after connection has completed. But still we had a case when we hit this select case and it after that when we log and try to reach the rmqErr.Reason or the first one we get a nil pointer dereference.

Reply:
Probably the reason can be `RMQ closed the connection while trying to reconnect with nil rmqErr` But the rmqErr can be nil on other cases also, that is why I didn't put it in the first place.